src/output-json.c

```diff
@@ -148,6 +148,15 @@ json_t *SCJsonString(const char *val)
 /* Default Sensor ID value */
 static int64_t sensor_id = -1; /* -1 = not defined */
 
+json_t *JsonAddStringN(const char *string, size_t size)
```

Comment:
Can you document this function?

Also, it relies on the caller to make sure that the desired size is not actually bigger than the string, or memcpy will copy data from beyond string. If string is really a string, we could compare size against strlen(string) to make sure this can't happen.

Reply:
This function is designed to convert a character sequence (not necessarily a C-string) into a JSON object. I will add documentation.